### PR TITLE
BUGFIX: Fix ResourceTypeConverter for "data" import

### PR DESCRIPTION
**Pull request by gerdemann**
Fix the ResourceTypeConverter so that you 
can also import resources via base64.

**What I did**
If one wanted to convert a base64 file via "data", this did not work.

**How I did it**
I changed the default value from "resource" to "null" so that the condition fits again.

**How to verify it**
Convert a resource based on an array with "data" and "filename".

**Checklist**

- [x] Code follows the PSR-2 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)

### CODE REVIEW

**Comment by sorenmalling** [comment]
Hi @gerdemann 😃 
Thanks for the pull request 👍 

Will you be so kind and translate it to english? And can you provide a test case?

**Comment by gerdemann** [comment]
Sorry for the german text, that was not intentional. 🙈

I can see tomorrow if I can create the missing tests.

**Comment by gerdemann** [comment]
Hi @sorenmalling,

I have tried to add a test. Can you please look at it again? 😄

File: Neos.Flow/Tests/Unit/ResourceManagement/ResourceTypeConverterTest.php [hunk 1]
```diff
@@ -92,7 +92,22 @@ public function convertFromReturnsNullIfSourceArrayIsEmpty()
     public function convertFromReturnsNullIfNoFileWasUploaded()
     {
         $source = ['error' => \UPLOAD_ERR_NO_FILE];
-        $this->assertNull($this->resourceTypeConverter->convertFrom($source, PersistentResource::class));
+        $this->assertTrue($this->resourceTypeConverter->convertFrom($source, PersistentResource::class));
```
Comment (sorenmalling) [inline Neos.Flow/Tests/Unit/ResourceManagement/ResourceTypeConverterTest.php:95]: Why was this test result changed? Did your change break it?

**Review by sorenmalling — CHANGES_REQUESTED** [review]

File: Neos.Flow/Tests/Unit/ResourceManagement/ResourceTypeConverterTest.php (hunk 1 above)
Comment (gerdemann) [reply Neos.Flow/Tests/Unit/ResourceManagement/ResourceTypeConverterTest.php]: Sorry, that should not be committed with, is changed now.

**Comment by kdambekalns** [comment]
When did this break, I wonder?

**Comment by gerdemann** [comment]
> When did this break, I wonder?

This has been changed from version 2.2 to 2.3 and has been broken since then.

**Review by albe — APPROVED** [review]
I was unsure about the new requirement to specify a filename, but it kind of makes sense, because else the media type also can not be inferred. Looking good from my side, apart from the fact that 4.3 is security-only by now and hence this should target 5.3

**Comment by sorenmalling** [comment]
@gerdemann Care to change it to target 5.3 - then we can get it in :-)

**Comment by gerdemann** [comment]
@sorenmalling Sorry, I had lost sight of that again. I have now adjusted the branch.

**Comment by albe** [comment]
@sorenmalling can you remove your veto? Then I would merge this

**Review by sorenmalling — APPROVED** [review]

**Pull request merged**